File: 0283-move-zeroes/0283-move-zeroes.cpp
class Solution {
public:
    void moveZeroes(vector<int>& nums) {
        int count = 0;
        
        for (int i=0; i<nums.size(); ++i) {
            if (!nums[i]) {
                ++count;
                nums.erase(nums.begin()+i);
                --i; // move index back bc erased element
            }
        }
        
        for (int i=0; i<count; ++i) 
            nums.push_back(0);
    }
};

File: 0283-move-zeroes/0283-move-zeroes.py
class Solution:
    def moveZeroes(self, nums: List[int]) -> None:
        count = 0
        i = 0
        
        while i < len(nums):
            if nums[i] == 0:
                count += 1
                nums.pop(i)
                i -= 1
            i += 1
        
        for i in range(count):
            nums.append(0)
            
        return nums
        